File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        first, second, third = 0, 1, 1
        
        for _ in range(3, n + 1):
            first, second, third = second, third, first + second + third
            
        return first if n == 0 else third

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int first = 0; int second = 1; int third = 1;
        
        for (auto x = 3; x < (n + 1); x++)  {
            int temp = first + second + third;
            first = second;
            second = third;
            third = temp;
        }
    
        return (n == 0) ? first : third;
    }
};